### DISCUSSION TITLE
Questions about integrating @walletconnect/modal-react-native

### DISCUSSION THREAD

**Discussion by judeProground** [WalletConnect/walletconnect-monorepo#3331] [Developer Support] [answered]
Hello, I'm new to this web3 community. 👋 
I'm trying to support Metamask in our dApp(React-Native) using WalletConnect (not directly using MetaMask mobile sdk), because I want to expand supported wallets in the future. So, I'm using @walletconnect/modal-react-native as mentioned in the [documentation](https://docs.walletconnect.com/2.0/reactnative/walletconnectmodal/about), and I'm curious if it's okay to patch those packages and use some custom designs instead of the modal component provided by this package. Is this permitted according to WalletConnect's policies?

**Comment by glitch-txs** [comment, marked as answer]
Yes, we have more low level tools for these cases. Check sign API or universal provider, both are available.

https://docs.walletconnect.com/2.0/api/sign

https://docs.walletconnect.com/2.0/web/providers/universal

**Reply by judeProground** [reply]
Thanks for answering!
